### PR TITLE
Remove glymur pins for infra, and exclude 0.9.0 in setup.cfg

### PR DESCRIPTION
**Pull request by Cadair**
Round two.

### CODE REVIEW

**Comment by nabobalis** [comment]
U BREAK

**Comment by Cadair** [comment]
Oh fun. Looks like some other unrelated things have changed in Glymur 0.9 which need patching.

**Comment by nabobalis** [comment]
```
    if any((glymur.lib.openjp2.OPENJP2, glymur.lib.openjpeg.OPENJPEG)):
E   AttributeError: module 'glymur.lib' has no attribute 'openjpeg'
```

**Comment by dstansby** [comment]
Maybe add the exclusions everywhere, and then we can add support for glymur 0.9 in `2.0`? (and maybe backport to minor releases too)

**Comment by dstansby** [comment]
I might(?) have fixed it.

File: sunpy/tests/helpers.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
     SKIP_GLYMUR = True
 else:
     # See if we have a C backend
-    if any((glymur.lib.openjp2.OPENJP2, glymur.lib.openjpeg.OPENJPEG)):
+    if glymur.lib.openjp2.OPENJP2:
```
Comment (dstansby) [inline sunpy/tests/helpers.py]: Unless we want to support testing with glymur<0.9 when openjp2 doesn't work but openjpeg does, this change should fix things.

**Comment by dstansby** [comment]
Fun times: `RuntimeError: You must have a version of OpenJPEG at least as high as 2.3.0 before you can read JPEG2000 images with glymur.  Your version is 0.0.0`

File: sunpy/tests/helpers.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/tests/helpers.py]: Does this chance actually mean we have a min version of 0.9.1 for sunpy now?

**Comment by dstansby** [comment]
@wtbarnes why the approval, there's still a broken test 😆

File: sunpy/tests/helpers.py (hunk 1 above)
Comment (dstansby) [reply sunpy/tests/helpers.py]: Or we have a min version of openjpeg of 0.2.3, which was released 4th Oct 2017? (http://www.openjpeg.org/)

Comment (nabobalis) [reply sunpy/tests/helpers.py]: I'm not sure how we enforce that, which we can do with python packages?

Comment (dstansby) [reply sunpy/tests/helpers.py]: Sorry I got a bit confused, the requirement I suggested above is only for running the tests. Users should be able to go on using openjpeg<2 with glymur<0.9 if they want.

Alternatively we could add some fancier conditional logic here based on `LooseVersion(glymur)` to handle <0.9 and >0.9.

Comment (nabobalis) [reply sunpy/tests/helpers.py]: I also got confused, I didn't notice this was a test file.

Comment (Cadair) [reply sunpy/tests/helpers.py]: This is to work out if we should skip the openjpeg tests if glymur is installed but openjpeg isn't

**Comment by Cadair** [comment]
**DO NOT MERGE**

**Comment by nabobalis** [comment]
> **DO NOT MERGE**

You don't control me.

**Comment by wtbarnes** [comment]
@dstansby oops. For some reason I thought those failures were unrelated

**Pull request merged**